spec/controllers/items_controller_spec.rb

```diff
@@ -124,6 +124,26 @@
     end
   end
 
+  describe "PATCH #reserve" do
```

Comment:
I'd argue that `PUT` method is a better fit for this. `PUT` has to be idempotent, and updating the reserve attribute is. `PATCH` doesn't have this requirement so is a looser fit then `PUT` in this specific case imo.